### PR TITLE
feat(developer): Mention PHP 8.4 support

### PR DESCRIPTION
**Pull request by nickvergessen**
grafik

### CODE REVIEW

File: developer_manual/app_publishing_maintenance/app_upgrade_guide/upgrade_to_30.rst [hunk 1]
```diff
@@ -145,12 +145,43 @@ In this release support for PHP 8.0 was removed. Follow the steps below to make
...
+
+In this release support for PHP 8.4 was added. Follow the steps below to make your app compatible.
+
+1. If ``appinfo/info.xml`` has a dependency specification for PHP, increase the ``max-version`` to 8.4.
+
+.. code-block:: xml
+
+  <dependencies>
+    <php min-version="8.1" max-version="8.4" />
+    <nextcloud min-version="26" max-version="28" />
```
Comment (come-nc) [inline developer_manual/app_publishing_maintenance/app_upgrade_guide/upgrade_to_30.rst:167]:
```suggestion
    <nextcloud min-version="26" max-version="30" />
```
Makes no sense to support 8.4 but not 30.

File: developer_manual/app_publishing_maintenance/app_upgrade_guide/upgrade_to_31.rst [hunk 2]
```diff
@@ -65,6 +65,35 @@ Removed APIs
...
+
+In this release support for PHP 8.4 was added. Follow the steps below to make your app compatible.
+
+1. If ``appinfo/info.xml`` has a dependency specification for PHP, increase the ``max-version`` to 8.4.
+
+.. code-block:: xml
+
+  <dependencies>
+    <php min-version="8.1" max-version="8.4" />
+    <nextcloud min-version="26" max-version="28" />
```
Comment (come-nc) [inline developer_manual/app_publishing_maintenance/app_upgrade_guide/upgrade_to_31.rst:79]:
```suggestion
    <nextcloud min-version="26" max-version="31" />
```

File: developer_manual/app_publishing_maintenance/app_upgrade_guide/upgrade_to_30.rst (hunk 1 above)
Comment (nickvergessen) [reply developer_manual/app_publishing_maintenance/app_upgrade_guide/upgrade_to_30.rst:167]: Sorry, was copy pasta from adding 8.3 and seems I didnt replace all.
```suggestion
    <nextcloud min-version="28" max-version="30" />
```

**Review by come-nc — CHANGES_REQUESTED** [review]

File: developer_manual/app_publishing_maintenance/app_upgrade_guide/upgrade_to_31.rst (hunk 2 above)
Comment (nickvergessen) [reply developer_manual/app_publishing_maintenance/app_upgrade_guide/upgrade_to_31.rst]: Not sure either, but hopefully now

**Review by come-nc — APPROVED** [review]

**Pull request merged**